src/osx/carbon/font.cpp

```diff
@@ -949,86 +969,151 @@ bool wxNativeFontInfo::FromString(const wxString& s)
 #endif // __LP64__/!__LP64__
 
-    token = tokenizer.GetNextToken();
-    if ( !token.ToLong(&l) )
-        return false;
-    m_family = (wxFontFamily)l;
-
-    token = tokenizer.GetNextToken();
-    if ( !token.ToLong(&l) )
-        return false;
-    m_style = (wxFontStyle)l;
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        m_family = (wxFontFamily)l;
 
-    token = tokenizer.GetNextToken();
-    if ( !token.ToLong(&l) )
-        return false;
-    m_ctWeight = WXWeightToCT(wxFont::ConvertFromLegacyWeightIfNecessary(l));
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        m_style = (wxFontStyle)l;
 
-    token = tokenizer.GetNextToken();
-    if ( !token.ToLong(&l) )
-        return false;
-    m_underlined = l != 0;
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        m_ctWeight = WXWeightToCT(wxFont::ConvertFromLegacyWeightIfNecessary(l));
 
-    if ( version == 0L )
-    {
-        m_strikethrough = false;
-    }
-    else
-    {
         token = tokenizer.GetNextToken();
         if ( !token.ToLong(&l) )
             return false;
-        m_strikethrough = l != 0;
-    }
+        m_underlined = l != 0;
 
-    // this works correctly via fallback even if this is (backwards compatibility) a font family name
-    SetPostScriptName( tokenizer.GetNextToken() );
-    
-    RealizeResource();
+        if ( version == 0L )
+        {
+            m_strikethrough = false;
+        }
+        else
+        {
+            token = tokenizer.GetNextToken();
+            if ( !token.ToLong(&l) )
+                return false;
+            m_strikethrough = l != 0;
+        }
+
+        // this works correctly via fallback even if this is (backwards compatibility) a font family name
+        SetPostScriptName( tokenizer.GetNextToken() );
+
+        RealizeResource();
 
 #ifndef __WXMAC__
-    if( !m_familyName )
-        return false;
+        if( !m_familyName )
+            return false;
 #endif
 
-    token = tokenizer.GetNextToken();
-    if ( !token.ToLong(&l) )
-        return false;
-    m_encoding = (wxFontEncoding)l;
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        m_encoding = (wxFontEncoding)l;
+        return true;
+    }
+    else if ( version == 2 )
+    {
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        bool underlined = l != 0;
 
-    return true;
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        bool strikethrough = l != 0;
+
+        token = tokenizer.GetNextToken();
+        if ( !token.ToLong(&l) )
+            return false;
+        wxFontEncoding encoding = (wxFontEncoding)l;
+
+        wxString xml = tokenizer.GetString();
+        xml+=GetPListPrefix();
```

Comment:
Shouldn't this be `xml = GetPlistPrefix() + xml;` ?

Reply:
of course, thanks a lot, how embarrassing ....